app/src/main/java/org/oppia/app/player/content/ContentListFragmentPresenter.kt

```diff
@@ -0,0 +1,57 @@
+package org.oppia.app.player.content
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.oppia.app.databinding.ContentListFragmentBinding
+import org.oppia.app.viewmodel.ViewModelProvider
+import org.oppia.data.backends.gae.model.GaeSubtitledHtml
+import org.oppia.domain.exploration.ExplorationProgressController
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+
+/** Presenter for [ContentListFragment]. */
+class ContentListFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<ContentViewModel>,
+  private val logger: Logger
+) {
+
+  private val entity_type: String = "exploration"
```

Comment:
Should this ``exploration`` value be passed in as an argument to the fragment? Otherwise how will we host this content list in non-exploration contexts?

Reply:
Beyond that, what are other values that this might be other than 'exploration'?

Reply:
> Should this `exploration` value be passed in as an argument to the fragment? Otherwise how will we host this content list in non-exploration contexts?

yes we need to pass it as argument in fragment. Corrected it.

Reply:
> Beyond that, what are other values that this might be other than 'exploration'?

I am not sure about this. As this is from exploration it has entityType `exploration`, this is used in forming image url. Like that, we have Skill, Train mode so image url may change depending on type.